### ISSUE TITLE
Errors in remove-useless-else and remove-iife

### ISSUE THREAD

**Issue by sad-dev** [coderaiser/putout#127]
```js
function foo(x) {
  return function () {
    if (x === 0) {
      return 1;
    } else {
      return function () {
        if (x === 1) {
          return 1;
        } else {
          return 2;
        }
      }();
    }
  }();
}
foo(1);
```

fails with 
```
3:4  error   Cannot convert undefined or null to object  remove-useless-else (parser)
```
and after setting ``"remove-useless-else": "off"``
```
6:13  error   Cannot convert undefined or null to object  remove-iife (parser)
```

**Comment by coderaiser** [comment]
Fixed with `@putout/plugin-remove-iife@3.0.0`, now it's transformed to:


```js
'use strict';

function foo(x) {
    if (x === 0) {
        return 1;
    }

    if (x === 1) {
        return 1;
    }

    return 2;
}
foo(1);
```

Landed in 🐊**Putout** [v28.6.0](https://github.com/coderaiser/putout/releases/tag/v28.6.0). Is it works for you?

**Comment by sad-dev** [comment]
Thanks for the quick reply! It fixes most of the cases, although I now have another error:

```
'use strict';
function foo(x) {
  (function () {
      return function () {
        if (x === 2) {
          return 2;
        } else {
          return 3;
        }
      }(), x += 1;
  })();
}
foo(1);
```

``4:13  error   Cannot read properties of null   remove-iife (parser)``

**Comment by coderaiser** [comment]
How it should be reduced?

```js
function foo(x) {
  (function () {
    (function() {
      return 2;
    })();
    return ++x;
  })();
}
foo(1);
```

First iteration:

```js
function foo(x) {
  (function () {
    2;
    return ++x;
  })();
}
foo(1);
```

Second iteration:

```js
function foo(x) {
 ++x;
}
foo(1);
```

This code have very little sense and can be just removed :)...
Did I miss something?

**Comment by coderaiser** [comment]
About `SequenceExpressions` support:

```js
'use strict';
function foo(x) {
  (function () {
      return function () {
        if (x === 2) {
          return 2;
        } else {
          return 3;
        }
      }(), x += 1;
  })();
}
foo(1);
```

```
4:13  error   Cannot read properties of null   remove-iife (parser)
```

Landed in `@putout/plugin-remove-iife v3.1.0` please re-install 🐊**Putout**.

Now fixed code looks this way:

```js
'use strict';

function foo(x) {
    (function() {
        (function() {
            if (x === 2) {
                return 2;
            }

            return 3;
        })();

        return ++x;
    })();
}
foo(1);
```

Is it works for you?

**Comment by sad-dev** [comment]
Yes, thank you. I'll make a new issue if more IIFE problems crop up
